### PR TITLE
Ci/components test app

### PR DESCRIPTION
**Pull request by tore-espressif**
# ESP-BSP Pull Request checklist

> Note: For new BSPs create a PR with this [link](https://github.com/espressif/esp-bsp/compare/main...my-branch?quick_pull=1&template=pr_template_bsp.md).
- [ ] Version of modified component bumped
- [x] CI passing

# Change description
_Please describe your change here_

### CODE REVIEW

**Comment by tore-espressif** [comment]
@hfudev Here we are adding idf-build-apps feature that checks modified files in the PR. Seems to be working well.

Could you please have a look whether there is something we can improve?

**Review by espzav — APPROVED** [review]
@tore-espressif Thank you for this PR! Good work. LGTM, please wait for review from another person :-)

File: .github/workflows/build-applications.yml [hunk 1]
```diff
@@ -48,20 +45,24 @@ jobs:
...
-          export IDF_EXTRA_ACTIONS_PATH=${GITHUB_WORKSPACE}/examples
           . ${IDF_PATH}/export.sh
-          pip install idf-component-manager==1.* ruamel.yaml idf-build-apps --upgrade
-          echo idf-build-apps extra arguments: ${{ env.extra_arguments }}
-          idf-build-apps find ${{ env.extra_arguments }} -p examples
-          idf-build-apps build --parallel-count ${{ matrix.parallel_count }} --parallel-index ${{ matrix.parallel_index }} ${{ env.extra_arguments }} -p examples
+          pip install idf-component-manager==1.* ruamel.yaml idf-build-apps==2.4.3 --upgrade
+
+          echo "Files changed: ${ALL_CHANGED_FILES}"
+          idf-build-apps find --modified-files "${ALL_CHANGED_FILES}"
```
Comment (hfudev) [inline .github/workflows/build-applications.yml:67]: no need to find first.

**Review by hfudev — APPROVED** [review]
LGTM!

**Review by hfudev — APPROVED** [review]
LGTM!

File: .github/workflows/build-applications.yml (hunk 1 above)
Comment (tore-espressif) [reply .github/workflows/build-applications.yml:67]: Thanks! I like to have the output printed in the CI, just for me to double check that all jobs that I want are selected :)

**Pull request merged**